include/nbl/builtin/glsl/ext/FFT/fft.glsl

```diff
@@ -0,0 +1,244 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_GLSL_EXT_FFT_INCLUDED_
+#define _NBL_GLSL_EXT_FFT_INCLUDED_
+
+#include <nbl/builtin/glsl/math/complex.glsl>
+
+// Shared Memory
+#include <nbl/builtin/glsl/workgroup/shared_arithmetic.glsl>
+#include <nbl/builtin/glsl/math/functions.glsl>
+#include <nbl/builtin/glsl/ext/FFT/parameters.glsl>
+
+
+#ifndef _NBL_GLSL_EXT_FFT_MAX_DIM_SIZE_
+#error "_NBL_GLSL_EXT_FFT_MAX_DIM_SIZE_ should be defined."
+#endif
+
+#ifndef _NBL_GLSL_EXT_FFT_MAX_ITEMS_PER_THREAD
```

Comment:
btw is `_NBL_GLSL_EXT_FFT_MAX_ITEMS_PER_THREAD` required to be a power of two?

Reply:
seems like it,
the max dim is power of two
so _NBL_GLSL_EXT_FFT_MAX_ITEMS_PER_THREAD is power of two if WORKGROUP_SIZE is power of two
which it is